File: 05 SWEA/D2/Python/Prob2005.py

"""
title  : 2005. 파스칼의 삼각형 (D2)
time   : 110ms
memory : 44816kb
"""

def solution():

    T = int(input())
    for tc in range(1, T+1):

        N = int(input())
        number = [[1] * idx for idx in range(1, N+1)]

        for idx in range(2, N):
            for jdx in range(1, idx):
                number[idx][jdx] = number[idx-1][jdx-1] + number[idx-1][jdx]

        print(f'#{tc}')
        for idx in range(N):
            print(*number[idx], sep=" ")

solution()

File: 05 SWEA/D2/Java/src/Prob2005.java

/**
* title  : 2005. 파스칼의 삼각형 (D2)
* time   : 95ms
* memory : 18668kb
*/

import java.io.*;
 
public class Prob2005 {
    
    public static void main(String[] args) throws IOException {
        
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

        int T = Integer.parseInt(br.readLine());
        for (int tc = 1; tc <= T; tc++) {

            int N = Integer.parseInt(br.readLine());
            int[][] number = new int[N][N];

            for (int i = 0; i < N; i++) {
                number[i][0] = 1;
                number[i][i] = 1;
            }

            for (int i = 2; i < N; i++) {
                for (int j = 1; j < i; j++) {
                    number[i][j] = number[i-1][j-1] + number[i-1][j];
                }
            }

            sb.append("#" + tc).append("\n");
            for (int i = 0; i < N; i++) {
                for (int j = 0; j <= i; j++) {
                    sb.append(number[i][j] + " ");
                }
                sb.append("\n");
            }
        }

        System.out.println(sb);
    }
}
